### ISSUE TITLE
GetAvailableCredit Bug & GetBlocksToMaturity

### ISSUE THREAD

**Issue by digibyte** [digibyte/digibyte#17]
There is a bug in GetAvailableCredit that is causing the daemon to report balances incorrectly when there are immature blocks in the wallet (this is related to changing the minimum block maturity to 100 blocks):

wallet.h line 624 is: https://github.com/digibyte/DigiByteProject/blob/master/src/wallet.h#L624

   `if (IsCoinBase() && GetBlocksToMaturity((chainActive.Height() - GetDepthInMainChain()) > 0 ))` 
It should be:

  `if (IsCoinBase() && GetBlocksToMaturity(chainActive.Height() - GetDepthInMainChain()) > 0 )` 

Notice the difference in the parenthesis.  The first one is passing this value into GetBlocksToMaturity:
   (chainActive.Height() - GetDepthInMainChain()) > 0 
Because it contains the > 0, it is evaluating to true and passing true into GetBlocksToMaturity which is expecting an integer.

Instead it should pass chainActive.Height() - GetDepthInMainChain() to GetBlocksToMaturity and then compare the result to 0, similar to the way it is don in GetCredit and GetImmatureCredit.

The second bug is in wallet.cpp at line 1027:
https://github.com/digibyte/DigiByteProject/blob/master/src/wallet.cpp#L1027

  `if (pcoin->IsCoinBase() && pcoin->GetBlocksToMaturity(pcoin->GetDepthInMainChain()) > 0)` 
should be:

  `if (pcoin->IsCoinBase() && pcoin->GetBlocksToMaturity(chainActive.Height() - pcoin >GetDepthInMainChain()) > 0)` 

Again, GetBlocksToMaturity is expecting the height of the block to be considered, not the depth in the chain for the block.  This is causing inputs from newly minted blocks to be considered for inclusion in a transaction when they have not yet matured.  This results in the error message "error: {"code":-4,"message":"Error: The transaction was rejected! This might happen if some of the coins in your wallet were already spent, such as if you used a copy of wallet.dat and coins were spent in the copy but not marked as spent here."}" when someone tries to spend newly minted coins.

Thanks to raveneye for pointing this out!

**Comment by l0rdicon** [comment]
I have this in a pull request now, just needs to be merged into master.  The analysis is totally correct. 

https://github.com/digibyte/DigiByteProject/pull/19

**Comment by digibyte** [comment]
Error when compiling this pull request.

wallet.cpp:1027:90: error: invalid operands of types ‘int’ and ‘const CWalletTx*’ to binary ‘operator-’
wallet.cpp:1027:117: error: ‘GetDepthInMainChain’ was not declared in this scope
  CXX    walletdb.o

**Comment by nochowderforyou** [comment]
I pushed the wrong copy.   Missed a  -     on line 1027 in the version  I pushed. I miss the command line, no more code source for me.  

Will rebase and push in the morning.   And this is why pull requests are the way.  More then one person can test them before being merged. 

Don't mind me posting from this account, it's the one my tablet is hooked up to. Thus why I can't rebase at the moment :-)
